draft-ietf-jsonpath-base.md

```diff
@@ -1568,11 +1588,31 @@ If the nodelist contains one or more nodes, the result is
 `LogicalTrue`.
 If the nodelist is empty, the result is `LogicalFalse`.
 
+### `single` Function Extension {#single-fn}
+
+Parameters:
+: 1. `NodesType`
+
+Result:
+: `SingleNodeType`
+
+The "single" function extension provides a way to obtain a
+node in a nodelist and make that available for further
+processing in the filter expression:
+
+~~~ JSONPath
+$[?value(single(fn(@.*.price))) >= 7.99]
```

Comment:
No one is going to want to type this when `$[?fn(@.*.price) >= 7.99]` is just as understandable.  I fear we're defining this grammar into non-adoption.

Reply:
The point of this PR is to explore what it looks like to mirror (on the function side) the fact that expressions like `$[?@.*.price >= 7.99]` are not well-formed and to avoid what, in my opinion, is a surprise in #410 of `value` converting a multi-node nodelist to `Nothing`.

**If** (big if), we chose to go in this direction, we could abbreviate `value(single(...))` by providing another function (e.g. `uniqueValue`) which composes `value` with `single`.

If `$[?fn(@.*.price) >= 7.99]` is understandable (remember that "fn is a hypothetical function extension that takes and results in `NodesType`"), then presumably so is `$[?@.*.price >= 7.99]`, but the WG has failed to reach a consensus on what that might mean, so we have ruled it out syntactially.